server/routerlicious/packages/protocol-base/src/quorum.ts

```diff
@@ -229,9 +226,57 @@ export class QuorumProposals extends TypedEventEmitter<IQuorumProposalsEvents> i
             throw new Error("Can't propose in disconnected state");
         }
 
-        const deferred = new Deferred<void>();
-        this.localProposals.set(clientSequenceNumber, deferred);
-        return deferred.promise;
+        return new Promise<void>((resolve, reject) => {
+            // The sequence number that our proposal went pending, which the msn must advance past.
+            // If undefined, then it's not sequenced yet.
+            let targetSequenceNumber: number | undefined;
```

Comment:
It's probably nicer to leverage the pendingProposal itself rather than separately calculating the approval from the MSN.  I might look at that as an option.

Reply:
Did this in latest.